src/TriangleNodesProvisioner.cpp

```diff
@@ -288,4 +291,157 @@ namespace blitzdg {
+
+        index_matrix_type Fm = *Fmask.get();
+
+
+        real_vector_type faceR(NumFacePoints), faceS(NumFacePoints);
+        real_matrix_type V1D(NumFacePoints, NumFacePoints), V1Dinv(NumFacePoints,NumFacePoints);
+        real_matrix_type massEdgeInv(NumFacePoints, NumFacePoints);
+
+        real_matrix_type massEdge1(NumFacePoints, NumFacePoints), massEdge2(NumFacePoints,NumFacePoints),
+                         massEdge3(NumFacePoints, NumFacePoints);
+
+        // Face 1
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceR(i) = r(Fm(i, 0));
+
+        std::cout << faceR << std::endl;
+
+        Vandermonde.computeVandermondeMatrix(faceR, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        std::cout << massEdgeInv << std::endl;
+
+        Inverter.computeInverse(massEdgeInv, massEdge1);
+
+        std::cout << massEdge1 << std::endl;
+
+        E = 0.0*jj;
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=0; j < NumFacePoints; ++j) {
+                E(Fm(i,0),j) = massEdge1(i,j);
+            }
+        }
+
+        // Face 2
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceR(i) = r(Fm(i, 1));
+        
+        std::cout << Fm(Range::all(), 1) << std::endl;
+        std::cout << faceR << std::endl;
+
+
+        Vandermonde.computeVandermondeMatrix(faceR, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge2);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=NumFacePoints; j < 2*NumFacePoints; ++j) {
+                E(Fm(i,1),j) = massEdge2(i,j % NumFacePoints);
+            }
+        }
+
+        // Face 3.
+        for (index_type i=0; i < NumFacePoints; ++i)
+            faceS(i) = s(Fm(i, 2));
+        
+
+        std::cout << faceS << std::endl;
+
+        Vandermonde.computeVandermondeMatrix(faceS, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge3);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=2*NumFacePoints; j < 3*NumFacePoints; ++j) {
+                E(Fm(i,2),j) = massEdge3(i,j % NumFacePoints);
+            }
+        }
+
+        std::cout << "E: " << std::endl << E << std::endl; 
+
+        // Get the Vandermonde guy.
+        real_matrix_type V2D(NumLocalPoints, NumLocalPoints);
+        V2D = 0.0*jj;
+        computeVandermondeMatrix(NOrder, r, s, V2D);
+
+        MassInv = 0.0*jj;
+        MassInv = sum(V2D(ii,kk)*V2D(jj,kk), kk);
+
+        std::cout << "Minv:" << std::endl << MassInv << std::endl;
+
+        // Multiply by inverse mass matrix;
+        Liftref = sum(MassInv(ii,kk)*E(kk,jj),kk);
+
+        std::cout << "Lift:" << std::endl << Liftref << std::endl;
+    }
+
+    const real_matrix_type & TriangleNodesProvisioner::get_Lift() const {
+        return *Lift.get();
```

Comment:
A unique_ptr has very similar semantics to a pointer, so you should be able to return `*Lift`, which is a bit cleaner.